src/api-service/__app__/onefuzzlib/azure/image.py

```diff
@@ -21,12 +21,20 @@ def get_os(region: Region, image: str) -> Union[Error, OS]:
     client = get_compute_client()
     parsed = parse_resource_id(image)
     if "resource_group" in parsed:
-        try:
-            name = client.images.get(
-                parsed["resource_group"], parsed["name"]
-            ).storage_profile.os_disk.os_type.name
-        except (ResourceNotFoundError, CloudError) as err:
-            return Error(code=ErrorCode.INVALID_IMAGE, errors=[str(err)])
+        if parsed["type"] == "galleries":
```

Comment:
Thinking through this: I think we've been assuming that if the `"resource_group"` key is present, we'll at least have a _key_ in `parsed` for every return key described in the `parse_resource_id()` docs. Unfortunately, that's not valid, and per the very weak contract of our parameter type, we can actually get an _empty_ dict.

Example:
```python
>>> rid_provider = '/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/my-resource-group/providers/Microsoft.Web/sites/my-site-name'
>>> rid_no_provider = '/subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/my-resource-group'
>>> parse_resource_id(rid_provider)
{'children': '',
 'name': 'my-site-name',
 'namespace': 'Microsoft.Web',
 'resource_group': 'my-resource-group',
 'resource_name': 'my-site-name',
 'resource_namespace': 'Microsoft.Web',
 'resource_parent': '',
 'resource_type': 'sites',
 'subscription': '11111111-1111-1111-1111-111111111111',
 'type': 'sites'}
>>> parse_resource_id(rid_no_provider)
{'resource_group': 'my-resource-group',
 'subscription': '11111111-1111-1111-1111-111111111111'}
```

This is not a great situation, and we should probably add types (in a `Union`) to explicitly represent the different expected cases, and allow clearer dispatching at this site.

Reply:
(this can be done by us in a follow-up PR)

Reply:
Captured in #1464.